site/app/libraries/FileUtils.php

```diff
@@ -251,6 +251,31 @@ public static function getAllDirs($path) {
         return $return;
     }
 
+    /**
+     * Given a path and a text string, return all directories in the path that have names containing the text argument.
+     *
+     * @param string $path
+     * @param string $text
+     *
+     * @return array
+     */
+    public static function getDirWithText($path, $text) {
```

Comment:
I think a more elegant solution would be to retrieve all directories and then search the resulting array. It would also probably be faster and you wouldn't have to repeat all of the getAllDirs code.